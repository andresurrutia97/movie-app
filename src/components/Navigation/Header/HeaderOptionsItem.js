import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

const HeaderOptionsItem = ({ path, name }) => {
  return (
    <NavLink className="flex items-center ml-6" to={path} exact>
      <span>{name}</span>
    </NavLink>
  );
};

HeaderOptionsItem.propTypes = {
  path: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};

export default HeaderOptionsItem;
