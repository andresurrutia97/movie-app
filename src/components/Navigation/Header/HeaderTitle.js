import React from 'react';
import PropTypes from 'prop-types';

const HeaderTitle = ({ title }) => {
  return (
    <span className="font-semibold flex justify-center custom-blue-text text-2xl">
      {title}
    </span>
  );
};

HeaderTitle.propTypes = {
  title: PropTypes.string.isRequired,
};

export default HeaderTitle;
