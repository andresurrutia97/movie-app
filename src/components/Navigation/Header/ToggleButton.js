import React from 'react';
import PropTypes from 'prop-types';

const ToggleButton = ({ clicked }) => {
  return (
    <div className="block absolute left-0 md:hidden">
      <button
        onClick={clicked}
        type="button"
        className="flex items-center text-white"
      >
        <span className="material-icons">menu</span>
      </button>
    </div>
  );
};

ToggleButton.propTypes = {
  clicked: PropTypes.func.isRequired,
};

export default ToggleButton;
