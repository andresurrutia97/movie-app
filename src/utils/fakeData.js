export const moviesData = {
  page: 1,
  results: [
    {
      poster_path: '/e1mjopzAS2KNsvpbpahQ1a6SkSn.jpg',
      adult: false,
      overview:
        'From DC Comics comes the Suicide Squad, an antihero team of incarcerated supervillains who act as deniable assets for the United States government, undertaking high-risk black ops missions in exchange for commuted prison sentences.',
      release_date: '2016-08-03',
      genre_ids: [14, 28, 80],
      id: 297761,
      original_title: 'Suicide Squad',
      original_language: 'en',
      title: 'Suicide Squad',
      backdrop_path: '/ndlQ2Cuc3cjTL7lTynw6I4boP4S.jpg',
      popularity: 48.261451,
      vote_count: 1466,
      video: false,
      vote_average: 5.91,
    },
    {
      poster_path: '/lFSSLTlFozwpaGlO31OoUeirBgQ.jpg',
      adult: false,
      overview:
        'The most dangerous former operative of the CIA is drawn out of hiding to uncover hidden truths about his past.',
      release_date: '2016-07-27',
      genre_ids: [28, 53],
      id: 324668,
      original_title: 'Jason Bourne',
      original_language: 'en',
      title: 'Jason Bourne',
      backdrop_path: '/AoT2YrJUJlg5vKE3iMOLvHlTd3m.jpg',
      popularity: 30.690177,
      vote_count: 649,
      video: false,
      vote_average: 5.25,
    },
    {
      poster_path: '/hU0E130tsGdsYa4K9lc3Xrn5Wyt.jpg',
      adult: false,
      overview:
        'One year after outwitting the FBI and winning the publicâ€™s adulation with their mind-bending spectacles, the Four Horsemen resurface only to find themselves face to face with a new enemy who enlists them to pull off their most dangerous heist yet.',
      release_date: '2016-06-02',
      genre_ids: [28, 12, 35, 80, 9648, 53],
      id: 291805,
      original_title: 'Now You See Me 2',
      original_language: 'en',
      title: 'Now You See Me 2',
      backdrop_path: '/zrAO2OOa6s6dQMQ7zsUbDyIBrAP.jpg',
      popularity: 29.737342,
      vote_count: 684,
      video: false,
      vote_average: 6.64,
    },
    {
      poster_path: '/h28t2JNNGrZx0fIuAw8aHQFhIxR.jpg',
      adult: false,
      overview:
        'A recently cheated on married woman falls for a younger man who has moved in next door, but their torrid affair soon takes a dangerous turn.',
      release_date: '2015-01-23',
      genre_ids: [53],
      id: 241251,
      original_title: 'The Boy Next Door',
      original_language: 'en',
      title: 'The Boy Next Door',
      backdrop_path: '/vj4IhmH4HCMZYYjTMiYBybTWR5o.jpg',
      popularity: 22.279864,
      vote_count: 628,
      video: false,
      vote_average: 4.13,
    },
    {
      poster_path: '/vOipe2myi26UDwP978hsYOrnUWC.jpg',
      adult: false,
      overview:
        'An orphan boy is raised in the Jungle with the help of a pack of wolves, a bear and a black panther.',
      release_date: '2016-04-07',
      genre_ids: [12, 18, 14],
      id: 278927,
      original_title: 'The Jungle Book',
      original_language: 'en',
      title: 'The Jungle Book',
      backdrop_path: '/eIOTsGg9FCVrBc4r2nXaV61JF4F.jpg',
      popularity: 21.104822,
      vote_count: 1085,
      video: false,
      vote_average: 6.42,
    },
  ],
  total_results: 19629,
  total_pages: 982,
};

export const movieInfo = {
  adult: false,
  backdrop_path: '/fCayJrkfRaCRCTh8GqN30f8oyQF.jpg',
  belongs_to_collection: null,
  budget: 63000000,
  genres: [
    {
      id: 18,
      name: 'Drama',
    },
  ],
  homepage: '',
  id: 550,
  imdb_id: 'tt0137523',
  original_language: 'en',
  original_title: 'Fight Club',
  overview:
    'A ticking-time-bomb insomniac and a slippery soap salesman channel primal male aggression into a shocking new form of therapy. Their concept catches on, with underground "fight clubs" forming in every town, until an eccentric gets in the way and ignites an out-of-control spiral toward oblivion.',
  popularity: 0.5,
  poster_path: null,
  production_companies: [
    {
      id: 508,
      logo_path: '/7PzJdsLGlR7oW4J0J5Xcd0pHGRg.png',
      name: 'Regency Enterprises',
      origin_country: 'US',
    },
    {
      id: 711,
      logo_path: null,
      name: 'Fox 2000 Pictures',
      origin_country: '',
    },
    {
      id: 20555,
      logo_path: null,
      name: 'Taurus Film',
      origin_country: '',
    },
    {
      id: 54050,
      logo_path: null,
      name: 'Linson Films',
      origin_country: '',
    },
    {
      id: 54051,
      logo_path: null,
      name: 'Atman Entertainment',
      origin_country: '',
    },
    {
      id: 54052,
      logo_path: null,
      name: 'Knickerbocker Films',
      origin_country: '',
    },
    {
      id: 25,
      logo_path: '/qZCc1lty5FzX30aOCVRBLzaVmcp.png',
      name: '20th Century Fox',
      origin_country: 'US',
    },
  ],
  production_countries: [
    {
      iso_3166_1: 'US',
      name: 'United States of America',
    },
  ],
  release_date: '1999-10-12',
  revenue: 100853753,
  runtime: 139,
  spoken_languages: [
    {
      iso_639_1: 'en',
      name: 'English',
    },
  ],
  status: 'Released',
  tagline:
    "How much can you know about yourself if you've never been in a fight?",
  title: 'Fight Club',
  video: false,
  vote_average: 7.8,
  vote_count: 3439,
};
