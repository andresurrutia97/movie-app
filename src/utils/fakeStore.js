export const fakeStore = {
  movies: {
    popularMovies: null,
    loadingPopularMovies: false,
    errorPopularMovies: null,
    topRatedMovies: null,
    loadingTopRatedMovies: false,
    errorTopRatedMovies: null,
    latestMovie: null,
    loadingLatestMovie: false,
    errorLatestMovie: null,
  },
  movie: { movieInfo: null, loading: false, error: null },
  favorites: {
    favoritesList: null,
    loadingFavList: false,
    errorFavList: null,
    favorites: null,
    loadingFavs: false,
    errorFavs: null,
    addFavorite: null,
    loadingAddFav: false,
    errorAddFav: null,
    removeFavorite: null,
    loadingRemFav: false,
    errorRemFav: null,
  },
  seen: {
    seenList: null,
    loadingSeenList: false,
    errorSeenList: null,
    seen: null,
    loadingSeen: false,
    errorSeen: null,
    addSeen: null,
    loadingAddSeen: false,
    errorAddSeen: null,
  },
};
