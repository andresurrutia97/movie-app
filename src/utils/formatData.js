const notDefinded = 'N/D';

export const formatYear = (year, defaultValue = notDefinded) =>
  year?.split('-')[0] || defaultValue;

export const formatCountry = (country, defaultValue = notDefinded) =>
  country ? country[0]?.iso_3166_1 : defaultValue;
