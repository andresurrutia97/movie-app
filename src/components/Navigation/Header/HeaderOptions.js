import React from 'react';
import PropTypes from 'prop-types';

import HeaderOptionsItem from './HeaderOptionsItem';
import Dropdown from './UserDropdown/UserDropdown';

const HeaderOptions = ({ routes }) => {
  const options = routes.map(({ path, name }) => {
    return (
      <li key={path}>
        <HeaderOptionsItem path={path} name={name} />
      </li>
    );
  });

  return (
    <div className=" hidden md:ml-8 md:w-full md:block md:flex md:justify-between">
      <ul className="flex flex-col text-sm text-white md:flex-row md:items-center">
        {options}
      </ul>
      <Dropdown />
    </div>
  );
};

HeaderOptions.propTypes = {
  routes: PropTypes.arrayOf(
    PropTypes.shape({
      path: PropTypes.string,
      name: PropTypes.string,
    })
  ).isRequired,
};

export default HeaderOptions;
