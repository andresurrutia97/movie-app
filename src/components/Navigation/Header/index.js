import React, { useRef, useEffect } from 'react';
import PropTypes from 'prop-types';

import ToggleButton from './ToggleButton';
import HeaderTitle from './HeaderTitle';
import HeaderOptions from './HeaderOptions';
import Routes from '../../../routes';

const Header = ({ openSidebarHandler }) => {
  const [navBackground, setNavBackground] = React.useState(
    'custom-black-header-grad'
  );
  const navRef = useRef();

  useEffect(() => {
    const handleScroll = () => {
      const { scrollY } = window;
      if (scrollY >= 10) {
        setNavBackground('custom-black-header');
      } else {
        setNavBackground('custom-black-header-grad');
      }
    };
    document.addEventListener('scroll', handleScroll);
    return () => {
      document.removeEventListener('scroll', handleScroll);
    };
  }, []);

  return (
    <nav
      ref={navRef}
      className={`w-full h-16 fixed px-4 py-3 z-10 transition-colors duration-500 ${navBackground}`}
    >
      <div className="w-full flex justify-center items-center relative md:px-16 ">
        <ToggleButton clicked={openSidebarHandler} />
        <HeaderTitle title="Uruflix" />
        <HeaderOptions routes={Routes} />
      </div>
    </nav>
  );
};

Header.propTypes = {
  openSidebarHandler: PropTypes.func,
};

Header.defaultProps = {
  openSidebarHandler: () => {},
};

export default Header;
