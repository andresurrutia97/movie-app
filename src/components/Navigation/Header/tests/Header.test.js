import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import Header from '../index';
import * as ReduxHooks from '../../../../utils/reduxHooks';

describe('<Header />', () => {
  const component = (
    <BrowserRouter>
      <Header />
    </BrowserRouter>
  );
  let store;

  beforeEach(() => {
    store = configureStore([thunk])({});
    jest
      .spyOn(ReduxHooks, 'useSelector')
      .mockImplementation(() => store.getState());
  });

  it('should render correctly ', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
