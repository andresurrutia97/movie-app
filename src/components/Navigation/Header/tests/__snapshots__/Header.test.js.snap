// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render correctly  1`] = `
<nav
  className="w-full h-16 fixed px-4 py-3 z-10 transition-colors duration-500 custom-black-header-grad"
>
  <div
    className="w-full flex justify-center items-center relative md:px-16 "
  >
    <div
      className="block absolute left-0 md:hidden"
    >
      <button
        className="flex items-center text-white"
        onClick={[Function]}
        type="button"
      >
        <span
          className="material-icons"
        >
          menu
        </span>
      </button>
    </div>
    <span
      className="font-semibold flex justify-center custom-blue-text text-2xl"
    >
      Uruflix
    </span>
    <div
      className=" hidden md:ml-8 md:w-full md:block md:flex md:justify-between"
    >
      <ul
        className="flex flex-col text-sm text-white md:flex-row md:items-center"
      >
        <li>
          <a
            aria-current="page"
            className="flex items-center ml-6 active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            <span>
              Home
            </span>
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            className="flex items-center ml-6"
            href="/user"
            onClick={[Function]}
          >
            <span>
              Profile
            </span>
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            className="flex items-center ml-6"
            href="/toc"
            onClick={[Function]}
          >
            <span>
              ToC
            </span>
          </a>
        </li>
      </ul>
      <div
        className="relative"
      >
        <button
          className="relative z-10 block h-8 w-8 rounded-full overflow-hidden border-2 custom-blue-border focus:outline-none"
          onClick={[Function]}
          type="button"
        >
          <img
            alt="Your avatar"
            className="h-full w-full object-cover"
            src="default-user-image.png"
          />
        </button>
        <div
          className="hidden absolute right-0 mt-2 py-2 text-sm w-48 bg-white rounded custom-shadow"
        >
          <div
            className="flex justify-between itms-center text-gray-700 py-2 px-5 cursor-pointer"
            onClick={[Function]}
            role="presentation"
          >
            Profile
            <span
              className="material-icons"
            >
              person
            </span>
          </div>
          <div
            className="flex justify-between itms-center text-gray-700 py-2 px-5 cursor-pointer"
            onClick={[Function]}
            role="presentation"
          >
            <div>
              Logout
            </div>
            <span
              className="material-icons"
            >
              exit_to_app
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</nav>
`;
